scico/numpy/util.py

```diff
@@ -52,15 +50,6 @@ def ensure_on_device(
                 stacklevel=2,
             )
 
-        elif not isinstance(
-            array,
-            (DeviceArray, BlockArray, ShardedDeviceArray),
-        ):
-            raise TypeError(
-                "Each element of parameter arrays must be ndarray, DeviceArray, BlockArray, or "
-                f"ShardedDeviceArray; Argument {i+1} of {len(arrays)} is {type(arrays[i])}."
-            )
-
         arrays[i] = jax.device_put(arrays[i])
```

Comment:
This might be controversial. My logic: instead of listing the types we are okay with, implicitly be okay with any type that can have `device_put` called on it.

Reply:
This seems reasonable to me.